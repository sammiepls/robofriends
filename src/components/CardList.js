import React from "react";
import Card from "./Card";

const CardList = ({ robots }) =>
  robots.map((robot) => <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} />);

export default CardList;
