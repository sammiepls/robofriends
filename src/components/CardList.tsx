import React from "react";
import Card from "./Card";
import { Robot } from "redux/actions/requestActions";

interface CardListProps {
  robots: Robot[];
}

const CardList: React.FC<CardListProps> = ({ robots }) => {
  return (
    <>
      {robots.map((robot) => (
        <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} />
      ))}
    </>
  );
};

export default CardList;
