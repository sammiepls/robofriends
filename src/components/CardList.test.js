import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('should render CardList component', () => {
  const mockRobots = [
    {
      id:1,
      name:'John Snow',
      username: 'johnjohn',
      email:'john@gmail.com'
    }
  ]
  expect(shallow(<CardList
  robots={mockRobots}/>)).toMatchSnapshot()
})
