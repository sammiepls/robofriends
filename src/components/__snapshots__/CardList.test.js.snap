// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList component 1`] = `
<Card
  email="john@gmail.com"
  id={1}
  key="1"
  name="John Snow"
/>
`;
